p2p/conn/connection.go

```diff
@@ -645,9 +662,20 @@ FOR_LOOP:
 				break FOR_LOOP
 			}
 			if msgBytes != nil {
-				c.Logger.Debug("Received bytes", "chID", channelID, "msgBytes", msgBytes)
-				// NOTE: This means the reactor.Receive runs in the same thread as the p2p recv routine
-				c.onReceive(channelID, msgBytes)
+				msgLen := uint32(len(msgBytes))
+				bufferMaxSize := uint32(defaultRecvMessageCapacity)
+				if msgLen > bufferMaxSize {
+					bufferMaxSize = msgLen
+				}
```

Comment:
@mhofman
Will this ever be True at this place? 🤔 

The function `recvPacketMsg` at line no. 656 must already throw err (`received message exceeds available capacity`) if `msgLen > bufferMaxSize` was true.

So I think this condition is redundant OR I might be missing something.

Reply:
My understanding is that a channel can be configured with a different message size limit, which can be higher than the default. If such a message were to be received on the channel, the connection needs to allow it.

Reply:
Aah I see, `FillDefaults` conditionally fills Channel's `RecvMessageCapacity` with `defaultRecvMessageCapacity`. Got it!

Btw shouldn't we respect (or use) `channel.desc.RecvMessageCapacity` as our `bufferMaxSize` especially when it is set lesser than the default (`defaultRecvMessageCapacity`)?

Reply:
I believe defaultRecvMessageCapacity is rightly used as our goal is to keep total msgBytes in queue (at any instance) to be within 21Mb, right? Otherwise we'd be burning 1000*21 memory.

So I guess we can resolve the discussion. Things are clarified 👍